app/views/clients/show.html.erb

```diff
@@ -15,38 +15,67 @@
 
-  <tbody>
-    <% @locations.each do |location| %>
+<script src="/assets/es6-shim/es6-shim.min.js"></script>
+<script src="/assets/systemjs/dist/system-polyfills.js"></script>
 
-      <tr id="<%= location.urn %>" class="h-card">
-        <td class="p-uid" id="urn--<%= location.urn %>">
-          <%= location.urn %>
-        </td>
-        <td class="p-name" id="name--<%= location.name %>">
-          <%= location.name %><br/>
-          <em><%= location.try(:internal_branded_name) %></em>
-        </td>
-        <td class="p-tel p-tel-default" id="default_number--<%= fetch_phone_number("default", location.phone_numbers) %>">
-          <%= fetch_phone_number("default", location.phone_numbers) %>
-        </td>
-        <td class="p-tel p-tel-mobile" id="mobile_number--<%= fetch_phone_number("mobile", location.phone_numbers) %>">
-          <%= fetch_phone_number("mobile", location.phone_numbers) %>
-        </td>
-        <td><%= link_to 'Edit', edit_location_path(location), class: "btn btn-default" %></td>
-        <td><%#= link_to 'Destroy', location, method: :delete, data: { confirm: 'Are you sure?' }, class: "btn btn-link" %></td>
-      </tr>
+<script src="/assets/angular2/bundles/angular2-polyfills.js"></script>
+<script src="/assets/systemjs/dist/system.src.js"></script>
+<script src="/assets/rxjs/bundles/Rx.js"></script>
+<script src="/assets/angular2/bundles/angular2.dev.js"></script>
+<script src="/assets/angular2/bundles/http.dev.js"></script>
+<script src="/assets/angular2/bundles/router.dev.js"></script>
 
+<script>
+  /**
+   * This is the module loading setup... but above we pre-load a bunch of libraries that
+   * we know we'll need to cut down on the latency of requesting one module at a time (it is actually very slow
+   * even for a small app like this).
+   *
+   * The above "bundles" (they call them bundles because they package a bunch of modules into a single file
+   * so that they can be included easily) will added to the SystemJS context relative to the current URL.  So if
+   * the URL for this page is http://localhost:3000/reports/g5-c-1slhp2tc-compass-rock-real-estate/monthly_overview
+   * then the bundles will register their modules like
+   * ```
+   * http://localhost:3000/reports/g5-c-1slhp2tc-compass-rock-real-estate/angular2/core
+   *  // for `angular2/core`
+   * ```
+   *
+   * So, in order to be able to load modules from different (absolute, non-parent relative) locations we need to map them.
+   */
+  System.config({
+    map: {
+      'clients_show': '/assets/ng2/clients_show',
+      'pipes': '/assets/ng2/pipes'
+    },
+    packages: {
+      'clients_show': {
+        format: 'register',
+        defaultExtension: 'js'
+      },
+      'pipes': {
+        format: 'register',
+        defaultExtension: 'js'
+      }
+    }
+  });
+  // and to finish, let's boot the app!
+  System.import('clients_show/clients_show.boot');
+</script>
+<script type="text/javascript">
+  window.locations = [
```

Comment:
I did it this way to try to be as minimally invasive to the project as
possible.  The rails templates were just rendering that same data into the
DOM so I changed it to a javascript object that I could reference from ng2.

On Mon, Feb 15, 2016 at 10:20 AM, steveburkett <notifications@github.com>
wrote:

> In app/views/clients/show.html.erb
> <https://github.com/G5/g5-phone-number-service/pull/52#discussion_r52929927>
> :
>
> > +    packages: {
> > +      'clients_show': {
> > +        format: 'register',
> > +        defaultExtension: 'js'
> > +      },
> > +      'pipes': {
> > +        format: 'register',
> > +        defaultExtension: 'js'
> > +      }
> > +    }
> > +  });
> > +  // and to finish, let's boot the app!
> > +  System.import('clients_show/clients_show.boot');
> > +</script>
> > +<script type="text/javascript">
> > +  window.locations = [
>
> same ? here as ^, for locations
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/G5/g5-phone-number-service/pull/52/files#r52929927>.
>



-- 
Jonathan